java/src/main/java/ai/rapids/cudf/HostMemoryBuffer.java

```diff
@@ -74,9 +74,15 @@ protected boolean cleanImpl(boolean logErrorIfNotClean) {
       boolean neededCleanup = false;
       long origAddress = address;
       if (address != 0) {
-        UnsafeMemoryAccessor.free(address);
-        MemoryListener.hostDeallocation(length, id);
-        address = 0;
+        try {
+          UnsafeMemoryAccessor.free(address);
+          MemoryListener.hostDeallocation(length, id);
```

Comment:
The call to `hostDeallocation` has a guard aginst `length == 0` for the pinned case, not sure if it's needed here.

Reply:
It's not needed because the cleaner is only checking the address.  This is all about ensuring the resource appears to be cleaned despite exceptions, and clearing the address is all that's necessary to ensure that.

Also I believe there are plans to remove `MemoryListener` functionality now that there's a memory event handler to get more accurate readings than the prediction code can provide.